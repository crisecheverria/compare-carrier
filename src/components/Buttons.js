import React, { useState } from "react"
import PropTypes from "prop-types"

function Buttons({ sortBy }) {
  const [btnClick, setBtnClick] = useState("")

  function handleClick(e) {
    setBtnClick(e.target.name)
    sortBy(e.target.name)
  }

  return (
    <h2 className="subtitle">
      <button
        type="button"
        onClick={handleClick}
        name="cheapest"
        className={
          btnClick === "cheapest" ? "button is-active" : "button is-light"
        }
        data-testid="btn-cheapest"
      >
        Cheapest
      </button>
      <button
        type="button"
        onClick={handleClick}
        name="fastest"
        className={
          btnClick === "fastest" ? "button is-active" : "button is-light"
        }
        data-testid="btn-fastest"
      >
        Fastest
      </button>
    </h2>
  )
}

Buttons.propTypes = {
  sortBy: PropTypes.func.isRequired,
}

export default Buttons
