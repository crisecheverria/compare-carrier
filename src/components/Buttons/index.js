import React, { useState } from "react"
import PropTypes from "prop-types"

function Buttons({ sortBy }) {
  const [selectedOption, setSelectedOption] = useState("")

  function handleClick(e) {
    setSelectedOption(e.target.name)
    sortBy(e.target.name)
  }

  return (
    <h2 className="subtitle">
      {["cheapest", "fastest"].map((option) => (
        <button
          key={option}
          type="button"
          onClick={handleClick}
          name={option}
          className={
            selectedOption === option ? "button is-active" : "button is-light"
          }
          data-testid="sorting-options"
        >
          {option}
        </button>
      ))}
    </h2>
  )
}

Buttons.propTypes = {
  sortBy: PropTypes.func.isRequired,
}

export default Buttons
