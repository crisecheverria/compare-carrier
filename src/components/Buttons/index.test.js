import React from "react"
import { render } from "@testing-library/react"
import Buttons from "."

test("render sorting options from <Buttons /> component", () => {
  function sortByParameter() {}
  const { getAllByTestId, baseElement } = render(
    <Buttons sortBy={sortByParameter} />
  )
  const sortingOptions = getAllByTestId(/sorting-options/i)

  expect(sortingOptions[0].name).toBe("cheapest")
  expect(sortingOptions[1].name).toBe("fastest")
  expect(baseElement).toMatchSnapshot()
})
