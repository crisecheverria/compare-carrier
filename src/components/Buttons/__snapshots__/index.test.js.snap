// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render sorting options from <Buttons /> component 1`] = `
<body>
  <div>
    <h2
      class="subtitle"
    >
      <button
        class="button is-light"
        data-testid="sorting-options"
        name="cheapest"
        type="button"
      >
        cheapest
      </button>
      <button
        class="button is-light"
        data-testid="sorting-options"
        name="fastest"
        type="button"
      >
        fastest
      </button>
    </h2>
  </div>
</body>
`;
